python/ray/train/v2/xgboost/xgboost_trainer.py

```diff
@@ -126,7 +127,32 @@ def __init__(
         dataset_config: Optional[ray.train.DataConfig] = None,
         metadata: Optional[Dict[str, Any]] = _UNSUPPORTED,
         resume_from_checkpoint: Optional[Checkpoint] = None,
+        # TODO(justinvyu): [Deprecated] Legacy XGBoostTrainer API
+        label_column: Optional[str] = None,
+        params: Optional[Dict[str, Any]] = None,
+        num_boost_round: Optional[int] = None,
```

Comment:
Are these needed for the V2 API? These are not passed in from the V1 API.

Reply:
This is `train/v2/xgboost`, not `train/xgboost/v2`, so people who use `RAY_TRAIN_V2_ENABLED` would possibly be passing in the xgboost V1 API.  😂  😭 💀 🪦